File: gabe/plus_one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        
        i = len(digits) - 1
        
        digits[-1] += 1

        carry = 0

        for i in range(len(digits)-1, -1, -1):
            digits[i] += carry

            if digits[i] > 9:
                digits[i] = 0
                carry = 1
            else:
                carry = 0
                break

        if carry == 1:
            digits.insert(0, 1)
            
        return digits

File: brady/plus_one.java
class Solution {
  
    // O(n) time, O(n) space
    public int[] plusOne(int[] digits) {
        int sum;
        for (int i = digits.length-1; i >= 0; --i) {
            ++digits[i];
            if (digits[i] <= 9) return digits;
            digits[i] = 0;
        }

        if (digits[0] == 0) {
            int[] newDigits = new int[digits.length+1];
            newDigits[0] = 1;
            return newDigits;
        }

        return digits;
    }
}
